app/formats/json/LabelFormat.scala

```diff
@@ -7,6 +7,7 @@ import play.api.libs.json._
 import play.api.libs.functional.syntax._
 
 object LabelFormat {
+  case class LoadedLabels(loaded_labels: String)
```

Comment:
Can you switch this up to read in a `Seq[Int]` instead of a String? You might not even need to add anything to the formats file, actually. I think that on the back end you can do `request.body.validate[Seq[Int]]` and on the front end I think you could just do
```
data: JSON.stringify(loadedLabels),
```

Reply:
After looking around a bit more, I have a better idea. Since we are already switching to a POST request of the label_ids, we should just put all of the metadata into the `data` field. So the data type would look something more like
```
case class GalleryLabelsRequest(labelTypeId: Int, n: Int, severities: Option[Seq[Int]], tags: Option[Seq[String]], loadedLabels: Seq[Int])
```

This way, we can actually combine the `getLabelsByType()` and `getLabelsBySeveritiesAndTags()` functions into one function that just behaves differently based on the input!

You could add it to `GalleryTaskSubmissionFormats.scala` instead of `LabelFormat.scala`, though I'd rename it to `GalleryFormats.scala` at that point as well.

Similarly, `fetchLabelsByType()` and `fetchLabelsBySeverityAndTags()` could probably be merged into one function. It looks like they are identical except for defining the URL.